react-forms/my-app/src/AIGeneratedControlled.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, FormEvent } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+
+  // Dedicated event handlers for clarity. Moving the arrow functions to the top makes the code
+  // cleaner and more clear for someone else wanting to work on it.
+  const handleUsernameChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    setUsername(e.target.value);
+  };
+
+  const handlePasswordChange = (
+    e: React.ChangeEvent<HTMLInputElement>
+  ): void => {
+    setPassword(e.target.value);
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // Insert validation logic here if needed
+    console.log({ username, password });
+  };
+  // notice the of htmlFor below. This helps with screen readability?
+  return (
+    <form onSubmit={handleSubmit} noValidate>
+      <div>
+        <label htmlFor="username">Username:</label>
```

Comment:
Wrap the label around the input. It makes the code simpler, shorter, and more readable. For example, your input does not need an `id` if you wrap the label.